lib/chef/resource/dsc_resource.rb

```diff
@@ -31,6 +31,7 @@ def initialize(name, run_context)
         super
         @properties = {}
         @resource = nil
+        @allow_reboot = :nothing

```

Comment:
I see now you're trying to allow :reboot_now and :request_reboot. I'd probably change this something like reboot_action.